tests/e2e/leaseProcessPool.spec.ts

```diff
@@ -174,4 +174,29 @@ describe("LeaseProcessPool", () => {
     );
     expect(Math.max(...poolSizesDuringWork)).toEqual(maxPoolSize);
   });
+
+  it("should abort acquiring lease process by signal", async () => {
+    const pool = glm.lease.createLeaseProcessPool(proposalPool, allocation, { replicas: 1 });
+    const abortControler = new AbortController();
+    abortControler.abort();
+    await expect(pool.acquire(abortControler.signal)).rejects.toThrow("The signing of the agreement has been aborted");
+  });
+
+  it("should abort acquiring lease process by timeout", async () => {
+    const pool = glm.lease.createLeaseProcessPool(proposalPool, allocation, { replicas: 1 });
+    await expect(pool.acquire(1_000)).rejects.toThrow("Could not sign any agreement in time");
+  });
+
+  it("should finalize the lease process during execution", async () => {
+    const pool = glm.lease.createLeaseProcessPool(proposalPool, allocation, { replicas: 1 });
+    const leaseProcess = await pool.acquire();
+    const exe = await leaseProcess.getExeUnit();
+    return new Promise(async (res) => {
+      leaseProcess.events.on("finalized", async () => res(true));
+      setTimeout(() => leaseProcess.finalize(), 8_000);
+      await expect(exe.run("sleep 10 && echo Hello World")).rejects.toThrow(
```

Comment:
if the promise gets resolved before this rejects then jest won't catch the error, we should add `expect.assertions(1)` at the beginning of the test